src/activities/ActivityEvaluationStatusEntity.js

```diff
@@ -0,0 +1,90 @@
+import { Entity } from '../es6/Entity.js';
+import { Rels } from '../hypermedia-constants';
+
+const supportedActivityTypes = [
+	'assignment',
+	'topic',
+	'quiz',
+];
+
+export class ActivityEvaluationStatusEntity extends Entity {
+	/**
+	 * @return {string|null} The activity type or null if the entity doesn't not have an expected class.
+	 */
+	getActivityType() {
+		if (this._entity) {
+			for (const hmClass of supportedActivityTypes) {
+				if (this._entity.hasClass(hmClass)) {
+					return hmClass;
+				}
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * @returns {int|null} The number of assignments of this activity
+	 */
+	numAssigned() {
+		return this._entity ? Number(this._entity.properties.assigned) : null;
```

Comment:
I think I understand that this is supposed to be a class to wrap the basic entity. Is that accurate? If yes, should this have a constructor that just returns a null object if the activity type is not supported?

Reply:
The reason I returned null was to be consistent with some of the entities I looked at. There's no real use case yet since the component that'll use this is still WIP, but basically the component has to handle these things being null.

I definitely like the idea of defaulting to 0 if we don't have an entity! One thing that's a bit weird about it though is that if we do this we're almost lying about the situation, because the entity isn't actually loaded and we're saying "the entity says there are 0 assignments of this activity"